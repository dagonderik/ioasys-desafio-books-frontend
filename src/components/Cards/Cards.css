 .card {
    background: white;
} 

img {
    width: auto;
    height: 150px;
}

.parent {
    flex: 5;
    flex-wrap: 'wrap';
    flex-direction: 'row';
}
.child {
    flex-basis: '50%';
}