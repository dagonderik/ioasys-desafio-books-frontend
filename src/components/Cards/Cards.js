import React from 'react';
import './Cards.css';
import PropTypes from 'prop-types';


const Cards = ({authors, title, pageCount, publisher, published, imageUrl, onRouteChange, id}) => {
    return (
        <div data-testid="Cards" key={id} onClick={() => onRouteChange(id)}
         className= 'card tc white dib br3 pa3 ma2 pointer grow shadow-5 card' 
         style={{height: "auto", maxWidth: "500px", minWidth: "400px", textAlign: "left", margin: "10px"}}  >
            <img alt='' src={`${imageUrl}`} style={{display:"inline-block", height:"150px", align: "center",verticalAlign: "middle"}} />
            <div style={{display:"inline-block", verticalAlign: "middle", marginLeft: "10px"}}>
                <h2 className='black f4'>{title}</h2>
                <p className='red f7'>
                    {authors.map((user,i) => {
                    return( <>{authors[i]}<br/></>)
                    })}
                </p>
                <p  className='black-50 f7'> {pageCount} {' p√°ginas'} <br/>{publisher}<br/>{published}</p>
               
            </div>
        </div>
    );
}

Cards.propTypes = {
    /**
     * authors holds an array of all the authors of this book
     */
    authors: PropTypes.array,
    /**
     * title holds the book title
     */
    title: PropTypes.string,
    /**
     * pageCount holds the number of pages of this book
     */
    pageCount: PropTypes.number,
     /**
     * publisher holds the information about the publisher of the book
     */
    publisher: PropTypes.string,
    /**
     * published holds the information about the year in which the book was published
     */
    published: PropTypes.string,
     /**
     * imageUrl holds the link to the book cover image
     */
    imageUrl: PropTypes.string,
    /**
     * onRouteChange is a funtion that receives a value to determine which screen to show
     */
    onRouteChange: PropTypes.func,
     /**
     * id holds a number to identify the position of the book in the array
     */
    id: PropTypes.number
    
    
}

export default Cards;