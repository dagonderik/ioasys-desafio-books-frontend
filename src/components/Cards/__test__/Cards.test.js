import React from "react";
import ReactDOM from 'react-dom';
import Cards from './../Cards';
import renderer from "react-test-renderer";

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Cards authors={["a","b"]} 
    title="title" 
    pageCount={10}
    publisher="publisher"
    published={1}
    imageUrl="imageUrl"
    onRouteChange={()=>{}}
    id={1}></Cards>,div)
})

it("matches snapshot", () =>{
    const tree = renderer.create(<Cards authors={["a","b"]} 
    title="title" 
    pageCount={10}
    publisher="publisher"
    published={1}
    imageUrl="imageUrl"
    onRouteChange={()=>{}}
    id={1}></Cards>).toJSON();
    expect(tree).toMatchSnapshot();
})