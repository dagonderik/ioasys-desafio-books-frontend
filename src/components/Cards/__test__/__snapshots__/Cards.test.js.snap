// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="card tc white dib br3 pa3 ma2 pointer grow shadow-5 card"
  data-testid="Cards"
  onClick={[Function]}
  style={
    Object {
      "height": "auto",
      "margin": "10px",
      "maxWidth": "500px",
      "minWidth": "400px",
      "textAlign": "left",
    }
  }
>
  <img
    alt=""
    src="imageUrl"
    style={
      Object {
        "align": "center",
        "display": "inline-block",
        "height": "150px",
        "verticalAlign": "middle",
      }
    }
  />
  <div
    style={
      Object {
        "display": "inline-block",
        "marginLeft": "10px",
        "verticalAlign": "middle",
      }
    }
  >
    <h2
      className="black f4"
    >
      title
    </h2>
    <p
      className="red f7"
    >
      a
      <br />
      b
      <br />
    </p>
    <p
      className="black-50 f7"
    >
       
      10
       
       p√°ginas
       
      <br />
      publisher
      <br />
      1
    </p>
  </div>
</div>
`;
